components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx

```diff
@@ -0,0 +1,41 @@
+import { h, Fragment } from '@financial-times/x-engine';
+import { LiveBlogPost } from '@financial-times/x-live-blog-post';
+import { withActions } from '@financial-times/x-interaction';
+
+const liveBlogWrapperActions = withActions({
+	insertPost (post) {
+		return ({ posts }) => {
+			const updatedPosts = [ post, ...posts ];
+
+			return { posts: updatedPosts };
+		};
+	},
+
+	updatePost (updated) {
+		return ({ posts }) => {
+			const updatedPosts = posts.map(
+				post => post.postId === updated.postId ? updated : post
+			);
+
+			return { posts: updatedPosts };
+		};
+	},
+
+	deletePost (postId) {
+		return ({ posts }) => {
+			const updatedPosts = posts.filter((post) => post.postId !== postId);
+			return { posts: updatedPosts };
+		};
+	}
+});
```

Comment:
I'm wondering if there are pros or cons to creating new arrays when handling each event (as is happening here) vs modifying the existing posts array directly e.g.

- `insertPost` - Add `post` to the beginning of the `posts` array with [`Array.prototype.unshift()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift)
- `updatePost` - Find post with [`Array.prototype.findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) and then do `posts[index] = updated`
- `deletePost` - Find post with [`Array.prototype.findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) and then use [`Array.prototype.splice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) to remove it

The `posts` array could get quite large, so the main benefit I can think of when modifying the array in place is that you're not using more memory by creating copies of the posts array every time we receive a new event. I'm not sure how much this matters though :thinking: Interested to hear your thoughts on this!

Reply:
@simonplend Thanks for the suggestion. I have created PR for this: https://github.com/Financial-Times/x-dash/pull/517